include/cufinufft/impl.h

```diff
@@ -153,7 +155,13 @@ int cufinufft_makeplan_impl(int type, int dim, int *nmodes, int iflag, int ntran
       printf("[cufinufft] upsampfac automatically set to %.3g\n", d_plan->opts.upsampfac);
     }
   }
-
+  if (d_plan->opts.gpu_spreadinterponly) {
+    // XNOR implementation below with boolean logic.
+    if ((d_plan->opts.upsampfac !=1) == (type != 3)) {
```

Comment:
Open question. In type3 setting `gpu_spreadinterponly` to 1 has no effect basically, the transform will be done. Is there any meaning in having `gpu_spreadinterponly=2` internal for type 3 and `gpu_spreadinterponly=1` will also not do the nufft in type 3? We could also use a value bigger than 2 in case we will use 2 for something else. The user should never set 2.

Reply:
I think the way to reduce confusion here is: rename it to `gpu_t12spreadinterponly`, with the understanding that it affects type 1 or 2 only. It should either be 0 (default) or 1. It can be set to 1 by the type 1 or 2 user as long as upsampfac=1.0 is also set. Otherwise for type 3 it has no effect (and in the implementation of type 3 it can be overwritten).  By its new name, the type 3 user would not expect it so have an effect.   I believe this is the right logic that we want.  Would you guys be able to tweak it to that?

Reply:
Let me add here that this check is only required here now and not really needed in the underlying exec pipelines. This is just to ensure that we have upsampfac=1 if type!=3 when gpu_spreadinterponly is set.

Reply:
Great. I agree. no need to change the name of the opt.